carbon-design-system/carbon-addons-iot-react
# [DateTimePickerV2]: show error message when date is not select ( single select)

Closes #3691 

**Summary**

The changes are only for `DateTimePickerV2` single select cause we do not want to change behaviors for old `DateTimePickerV2` and only `single select` has the `clear` button which would cause date is not selected. 

A few changes here:

- Apply button is always enabled
- After click on apply button, we validate date/time.
-  If date not selected or time not selected/wrong format, we show error messages accordingly.
- DatePicker and TimePickerDropDown have their own error messages
- Top input show dateTimeMask if date or time not selected

**Change List (commits, features, bugs, etc)**

- DateTimePickerV2WithNewTimeSpinner : update onApplyClick() to validate date/time and then submit. 

**Acceptance Test (how to verify the PR)**

- Go to [single select ](https://deploy-preview-3777--carbon-addons-iot-react.netlify.app/?path=/story/2-watson-iot-experimental-%E2%98%A2%EF%B8%8F-datetimepickerv2--single-select)story
- Click calendar icon to open flyout menu
- Click clear button
- Click calendar icon again to open flyout menu
- Top input filed should display dateTimeMask in gray out. Date and Time are not selected. 
- Apply button is blue and enabled. 
- Click on Apply button. Error message should show under Date Picker and Time Picker
- Clear the value again and try time is selected but not date. Click Apply and error message should show under DatePicker only
- Clear the value and try date is selected but not time. Click Apply and error message should show under Time Picker.

https://user-images.githubusercontent.com/24279794/234600918-524c84c2-9a85-43f6-840d-fc9b19d7e6b6.mov



**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here



**Things to look for during review**
